Packs/PaloAltoNetworks_PAN_OS_EDL_Management/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py

```diff
@@ -263,37 +262,44 @@ def edl_update_external_file_command():
     edl_update_external_file(file_path, list_name, verbose)
 
 
-def edl_update_internal_list(list_name: str, list_items, add, verbose: bool):
+def edl_update_internal_list(list_name: str, list_items: list, add: bool, verbose: bool):
     dict_of_lists = demisto.getIntegrationContext()
+
     if not dict_of_lists:
+        demisto.debug(f'PAN-OS EDL Management integration context is empty.')
         dict_of_lists = {list_name: list_items}
         if verbose:
             md = tableToMarkdown('List items:', list_items, headers=[list_name])
         else:
             md = 'Instance context updated successfully.'
     else:
         if not dict_of_lists.get(list_name, None) and not add:
-            return_error('Cannot remove items from an empty list.')
+            raise Exception(f'Cannot remove items from an empty list: {list_name}.')
+
         if dict_of_lists.get(list_name, None):
             if add:
                 chosen_list = dict_of_lists.get(list_name)
                 if not isinstance(chosen_list, list):
                     chosen_list = [chosen_list]
 
                 list_items = list(set(chosen_list + list_items))
-            else:
+            else:  # remove
                 list_items = [item for item in dict_of_lists.get(list_name) if item not in list_items]
 
-        if len(list_items) == 0:  # delete list from instance context
+        if not add and len(list_items) == 0:
+            # delete list from instance context, can happen only upon remove of objects
+            demisto.debug(f'PAN-OS EDL Management deleting {list_name} from the integration context.')
             dict_of_lists.pop(list_name, None)
             md = 'List is empty, deleted from instance context.'
         else:
+            # update list in instance context, can happen upon removal or addition of objects
             dict_of_lists.update({list_name: list_items})
             if verbose:
                 md = tableToMarkdown('List items:', list_items, headers=[list_name])
             else:
                 md = 'Instance context updated successfully.'
 
+    demisto.debug(f'PAN-OS EDL Management updating {list_name} with {len(list_items)} in the integration context.')
```

Comment:
@bakatzir you are printing here the len(list_items), but you are not printing if the whole `dict_of_lists` is empty before you save it.

Reply:
Won't it be printed here?
`demisto.debug(f'PAN-OS EDL Management integration context is empty.')`